### PR TITLE
ST: Add test coverage for improvements for connect and connector operator

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Enhancement / new feature

### Description
Add system test coverage for https://github.com/strimzi/strimzi-kafka-operator/pull/2425 and https://github.com/strimzi/strimzi-kafka-operator/pull/2324

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectS2IST#testDeployS2IWithMongoDBPlugin,CustomResourceStatusST#testKafkaConnectAndConnectorStatus+testKafkaConnectorWithoutClusterConfig

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java [hunk 1]
```diff
@@ -250,6 +253,20 @@ void testKafkaConnectS2IStatus() throws InterruptedException {
         assertKafkaConnectS2IStatus(3, connectS2IUrl, connectS2IDeploymentConfigName);
     }
 
+    @Test
+    void testKafkaConnectorWithoutClusterConfig() {
+        // This test check NPE when connect cluster is not specified in labels
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: Are you saying that the NotReady has a `NullPointerException` as its reason? If so we should fix that.

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java [hunk 2]
```diff
@@ -416,13 +433,31 @@ void assertKafkaConnectStatus(long expectedObservedGeneration, String expectedUr
         KafkaConnectStatus kafkaConnectStatus = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus();
         assertThat("Kafka Connect cluster status has incorrect Observed Generation", kafkaConnectStatus.getObservedGeneration(), is(expectedObservedGeneration));
         assertThat("Kafka Connect cluster status has incorrect URL", kafkaConnectStatus.getUrl(), is(expectedUrl));
+
+        List<ConnectorPlugin> pluginsList = kafkaConnectStatus.getConnectorPlugins();
+        assertThat(pluginsList, notNullValue());
+        List<String> pluginsClasses = pluginsList.stream().map(p -> p.getConnectorClass()).collect(Collectors.toList());
+        assertThat(pluginsClasses, hasItems("org.apache.kafka.connect.file.FileStreamSinkConnector",
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: We should reuse the `checkConnectorInStatus()`.

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java [hunk 3]
```diff
@@ -416,13 +433,31 @@ void assertKafkaConnectStatus(long expectedObservedGeneration, String expectedUr
...
     void assertKafkaConnectS2IStatus(long expectedObservedGeneration, String expectedUrl, String expectedConfigName) {
         KafkaConnectS2Istatus kafkaConnectS2IStatus = KafkaConnectS2IResource.kafkaConnectS2IClient().inNamespace(NAMESPACE).withName(CONNECTS2I_CLUSTER_NAME).get().getStatus();
         assertThat("Kafka ConnectS2I cluster status has incorrect Observed Generation", kafkaConnectS2IStatus.getObservedGeneration(), is(expectedObservedGeneration));
         assertThat("Kafka ConnectS2I cluster status has incorrect URL", kafkaConnectS2IStatus.getUrl(), is(expectedUrl));
         assertThat("Kafka ConnectS2I cluster status has incorrect BuildConfigName", kafkaConnectS2IStatus.getBuildConfigName(), is(expectedConfigName));
+
+        List<ConnectorPlugin> pluginsList = kafkaConnectS2IStatus.getConnectorPlugins();
+        assertThat(pluginsList, notNullValue());
+        List<String> pluginsClasses = pluginsList.stream().map(p -> p.getConnectorClass()).collect(Collectors.toList());
+        assertThat(pluginsClasses, hasItems("org.apache.kafka.connect.file.FileStreamSinkConnector",
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: We should reuse the `checkConnectorInStatus()`.

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: No, it was already fixed and this test just check, that no NPE is in log during this test case.

**Comment by Frawless** [comment]
@tombentley Your suggestions are now in place

**Pull request merged**